packages/eslint-plugin/src/rules/no-empty-function.ts

```diff
@@ -74,30 +89,45 @@ export default util.createRule<Options, MessageIds>({
     }
 
     /**
-     * Checks if the method is a concise constructor (no function body, but has parameter properties)
      * @param node the node to be validated
-     * @returns true if the method is a concise constructor
+     * @returns true if the constructor is allowed to be empty
      * @private
      */
-    function isConciseConstructor(
+    function isAllowedEmptyConstructor(
       node: TSESTree.FunctionDeclaration | TSESTree.FunctionExpression,
     ): boolean {
-      // Check TypeScript specific nodes
-      return (
-        isConstructor(node) && isBodyEmpty(node) && hasParameterProperties(node)
-      );
+      const parent = node.parent;
+      if (
+        isBodyEmpty(node) &&
+        parent &&
+        parent.type === 'MethodDefinition' &&
```

Comment:
nit: optional chaining.
TS will refine the type of parent to be non-nullish for the rest of the scope.

```
parent?.type === 'MethodDefinition' &&
```

Reply:
Good point. Thanks.